### PR TITLE
Support OlapGroupBy in Query Builder

### PR DESCRIPTION
**Pull request by MauricioUyaguari**
## Summary

- [x] FIxes https://github.com/finos/legend-studio/issues/1527
- [x] minor cleanups to TDSState
- [x] Allow hiding of filter panel 

## How did you test this change?

- [x] Test(s) added
- [ ] Manual testing (please provide screenshots/recordings)
- [ ] No testing (please provide an explanation)

https://user-images.githubusercontent.com/29690824/196834709-4c7718c3-ad00-4616-bcf9-80a4a9348318.mp4

https://user-images.githubusercontent.com/29690824/196834717-64e88d6d-f880-4ea8-a723-4f9259bf7dfb.mp4

https://user-images.githubusercontent.com/29690824/196834729-bbbbcf7d-cdbf-4a92-ac8b-8587a9694091.mp4

### CODE REVIEW

**Review by akphi — APPROVED** [review]

File: packages/legend-query-builder/src/stores/fetch-structure/tds/olapGroupBy/QueryBuilderOlapGroupByState.ts [hunk 1]
```diff
@@ -0,0 +1,377 @@
...
+import { DEFAULT_LAMBDA_VARIABLE_NAME } from '../../../QueryBuilderConfig.js';
+import type { QueryBuilderProjectionColumnDragSource } from '../projection/QueryBuilderProjectionColumnState.js';
+import { QueryBuilderTDSColumnState } from '../QueryBuilderTdsColumnState.js';
+import type { QueryBuilderTDSState } from '../QueryBuilderTDSState.js';
+import type { COLUMN_SORT_TYPE } from '../QueryResultSetModifierState.js';
+import type { QueryBuilderTDSOlapOperator } from './operators/QueryBuilderTdsOlapOperator.js';
+
+export const QUERY_BUILDER_OLAP_COLUMN_DND_TYPE = 'OLAP_COLUMN';
+
+export interface QueryBuilderOlapColumnDragSource {
```
Comment (akphi) [inline packages/legend-query-builder/src/stores/fetch-structure/tds/olapGroupBy/QueryBuilderOlapGroupByState.ts:37]: TODO: let'probably should refactor this to support TDS column instead

File: packages/legend-query-builder/style/_query-builder-olap.scss [hunk 2]
```diff
@@ -0,0 +1,408 @@
...
+    width: 3.4rem;
+  }
+
+  &__column__action {
+    @include flexCenter;
+
+    height: 3.4rem;
+    width: 3.4rem;
+
+    .query-builder__icon,
```
Comment (akphi) [inline packages/legend-query-builder/style/_query-builder-olap.scss:172]: this looks weird, do you mean to do `.query-builder__icon` only, why do we need the `svg`?

File: packages/legend-query-builder/style/_query-builder-olap.scss [hunk 3]
```diff
@@ -0,0 +1,408 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+@use 'mixins' as *;
+
+.query-builder__olap {
```
Comment (akphi) [inline packages/legend-query-builder/style/_query-builder-olap.scss:19]: did we take care of `light theme`?

**Review by akphi — CHANGES_REQUESTED** [review]
Let's make sure we standardize `TDS` and `OLAP` all-caps. Thanks!
Also, let's fix up the pipeline.

**Review by akphi — APPROVED** [review]

**Comment by MauricioUyaguari** [comment]
###Service
Service model::OlapRank
{
  pattern: '/066507dc-6e4c-4cde-afbb-808151f26791';
  owners:
  [
    'anonymous'
  ];
  documentation: '';
  autoActivateUpdates: true;
  execution: Single
  {
    query: |model::Person.all()->project([x|$x.firstName, x|$x.lastName], ['First Name', 'Last Name'])->olapGroupBy(['First Name', 'Last Name'], asc('Last Name'), y|$y->rank(), 'testCol1');
    mapping: model::RelationalMapping;
    runtime:
    #{
      mappings:
      [
        model::RelationalMapping
      ];
      connections:
      [
        model::MyDatabase:
        [
          my_connection: model::MyConnection
        ]
      ];
    }#;
  }
}

Service model::RelationalMappingServiceSImple
{
  pattern: '/26e50858-ac0b-420c-bcc5-41e1bf3f9d0a';
  documentation: '';
  autoActivateUpdates: true;
  execution: Single
  {
    query: |model::product::Channel.all()->project([x|$x.name, x|$x.products.name, x|$x.products.price], ['Name', 'Products/Name', 'Price'])->olapGroupBy(['Name'], asc('Price'), 'Price'->func(x|$x->sum()), 'My Sum');
    mapping: model::RelationalMapping;
    runtime: model::Runtime;
  }
}

Service model::RelationalMappingServiceRegularAggregation
{
  pattern: '/60056c10-ff93-4efd-b5ed-2455b54ceb49';
  documentation: '';
  autoActivateUpdates: true;
  execution: Single
  {
    query: |model::product::Channel.all()->groupBy([x|$x.name, x|$x.products.name], [agg(x|$x.products.price, x|$x->sum())], ['Name', 'Products/Name', 'Products/Price']);
    mapping: model::RelationalMapping;
    runtime: model::Runtime;
  }
}


###Relational
Database model::MyDatabase
(
  Table FirmTable
  (
    id INTEGER PRIMARY KEY,
    Legal_name VARCHAR(200),
    Inc VARCHAR(200)
  )
  Table PersonTable
  (
    id INTEGER PRIMARY KEY,
    firm_id INTEGER,
    firstName VARCHAR(200),
    lastName VARCHAR(200),
    age INTEGER,
    date DATE
  )
  Table Channel
  (
    id INTEGER PRIMARY KEY,
    name VARCHAR(200)
  )
  Table Product
  (
    id INTEGER PRIMARY KEY,
    name VARCHAR(200),
    price INTEGER,
    channelId INTEGER
  )

  Join ChannelProduct(Product.channelId = Channel.id)
  Join FirmPerson(PersonTable.firm_id = FirmTable.id)
)


###Pure
Profile model::MyExtension
{
  stereotypes: [important];
  tags: [doc];
}

Enum model::IncType
{
  Corp,
  LLC
}

Class <<model::MyExtension.important>> {model::MyExtension.doc = 'This is a model of a firm'} model::Firm extends model::LegalEntity
[
  validName: $this.legalName->startsWith('_')
]
{
  employees: model::Person[1..*];
  incType: model::IncType[1];
  isApple: Boolean[1];
  employeeSize() {$this.employees->count()}: Number[1];
}

Class model::Person
{
  firstName: String[1];
  lastName: String[1];
  age: Integer[1];
  dob: Date[1];
  fullName() {$this.firstName + ' ' + $this.lastName}: String[1];
}

Class model::product::Channel
{
  name: String[1];
  products: model::product::Product[*];
}

Class model::product::Product
{
  name: String[1];
  price: Integer[1];
}

Class model::LegalEntity
{
  legalName: String[1];
}


###Mapping
Mapping model::RelationalMapping
(
  *model::Firm: Relational
  {
    ~primaryKey
    (
      [model::MyDatabase]FirmTable.id
    )
    ~mainTable [model::MyDatabase]FirmTable
    legalName: concat([model::MyDatabase]FirmTable.Legal_name, '_LTD'),
    employees[model_Person]: [model::MyDatabase]@FirmPerson,
    isApple: case(equal([model::MyDatabase]FirmTable.Legal_name, 'Apple'), 'true', 'false'),
    incType: EnumerationMapping model_IncType: [model::MyDatabase]FirmTable.Inc
  }
  *model::Person: Relational
  {
    ~primaryKey
    (
      [model::MyDatabase]PersonTable.id
    )
    ~mainTable [model::MyDatabase]PersonTable
    firstName: [model::MyDatabase]PersonTable.firstName,
    lastName: [model::MyDatabase]PersonTable.lastName,
    age: [model::MyDatabase]PersonTable.age,
    dob: [model::MyDatabase]PersonTable.date
  }
  *model::product::Product: Relational
  {
    ~primaryKey
    (
      [model::MyDatabase]Product.id
    )
    ~mainTable [model::MyDatabase]Product
    name: [model::MyDatabase]Product.name,
    price: [model::MyDatabase]Product.price
  }
  *model::product::Channel: Relational
  {
    ~primaryKey
    (
      [model::MyDatabase]Channel.id
    )
    ~mainTable [model::MyDatabase]Channel
    name: [model::MyDatabase]Channel.name,
    products[model_product_Product]: [model::MyDatabase]@ChannelProduct
  }

  model::IncType: EnumerationMapping
  {
    Corp: ['Corp', 'CORP'],
    LLC: ['LLC']
  }
)


###Connection
RelationalDatabaseConnection model::MyConnection
{
  store: model::MyDatabase;
  type: H2;
  specification: LocalH2
  {
    testDataSetupSqls: [
      'Drop table if exists FirmTable;\nDrop table if exists PersonTable;\nDrop table if exists Channel;\nDrop table if exists Product;\nCreate Table FirmTable(id INT, Legal_Name VARCHAR(200), Inc VARCHAR(200));\nCreate Table PersonTable(id INT, firm_id INT, lastName VARCHAR(200), firstName VARCHAR(200), age INT);\nCreate Table Channel(id INT, Name VARCHAR(200));\nCreate Table Product(id INT, Name VARCHAR(200), price INT, channelId INT);\nInsert into Channel (id, Name) values (1, \'Facebook\');\nInsert into Channel (id, Name) values (2, \'Google\');\nInsert into Channel (id, Name) values (3, \'Apple\');\nInsert into Channel (id, Name) values (4, \'Netflix\');\nInsert into Channel (id, Name) values (5, \'Microsoft\');\n\nInsert into Product (id, Name, price, channelId) values (1, \'T-Shirts\', 100, 1);\nInsert into Product (id, Name, price, channelId) values (2, \'Facebook Oculus\', 200, 1);\nInsert into Product (id, Name, price, channelId) values (3, \'Facebook Messegener\', 300, 1);\nInsert into Product (id, Name, price, channelId) values (4, \'Pixel\', 100, 2);\nInsert into Product (id, Name, price, channelId) values (5, \'Youtube\', 200, 2);\nInsert into Product (id, Name, price, channelId) values (6, \'MacBook\', 400, 3);\nInsert into Product (id, Name, price, channelId) values (7, \'Airpods\', 500, 3);\n\nInsert into Product (id, Name, price, channelId) values (8, \'Netflix Shows\', 600, 4);\nInsert into Product (id, Name, price, channelId) values (9, \'Netflix Music\', 600, 4);\nInsert into Product (id, Name, price, channelId) values (10, \'Netflix Video Games\', 600, 4);\n\nInsert into Product (id, Name, price, channelId) values (11, \'Outlook\', 700, 5);\nInsert into Product (id, Name, price, channelId) values (12, \'Teams\', 800, 5);\n\n\n\n\nInsert into FirmTable (id, Legal_Name, Inc) values (1, \'Finos\', \'CORP\');\nInsert into FirmTable (id, Legal_Name, Inc) values (2, \'Apple\', \'Corp\');\nInsert into FirmTable (id, Legal_Name, Inc) values (3, \'GS\', \'Corp\');\nInsert into FirmTable (id, Legal_Name, Inc) values (4, \'Google\', \'Corp\');\nInsert into FirmTable (id, Legal_Name, Inc) values (5, \'Alphabet\', \'LLC\');\nInsert into PersonTable (id, firm_id, lastName, firstName, age) values (1, 3, \'X1\', \'Mauricio\', 10);\nInsert into PersonTable (id, firm_id, lastName, firstName, age) values (2, 3, \'X2\', \'An\', 20);\nInsert into PersonTable (id, firm_id, lastName, firstName, age) values (3, 3, \'X3\', \'Anne\', 30);\nInsert into PersonTable (id, firm_id, lastName, firstName, age) values (4, 3, \'X4\', \'Gayathri\', 40);\nInsert into PersonTable (id, firm_id, lastName, firstName, age) values (5, 3, \'X5\', \'Yannan\', 50);\nInsert into PersonTable (id, firm_id, lastName, firstName, age) values (6, 3, \'X6\', \'Dave\', 60);\nInsert into PersonTable (id, firm_id, lastName, firstName, age) values (7, 3, \'X7\', \'Mo\', 70);\nInsert into PersonTable (id, firm_id, lastName, firstName, age) values (8, 3, \'X9\', \'Teddy\', 80);\nInsert into PersonTable (id, firm_id, lastName, firstName, age) values (9, 2, \'X8\', \'Teddy\', 90);\nInsert into PersonTable (id, firm_id, lastName, firstName, age) values (10, 2, \'X8\', \'Teddy\', 100);'
      ];
  };
  auth: DefaultH2;
}


###Runtime
Runtime model::Runtime
{
  mappings:
  [
    model::RelationalMapping
  ];
  connections:
  [
    model::MyDatabase:
    [
      my_connection: model::MyConnection
    ]
  ];
}

**Pull request merged**